src/App.tsx

```diff
@@ -16,12 +18,27 @@ function App() {
           <Route path="/">
             <Route index element={<MainPage />} />
             <Route path="chapters">
-              <Route path="making-your-first-chart" element={<LineChart />} />
-              <Route path="making-a-scatterplot" element={<ScatterPlot />} />
-              <Route path="making-a-bar-chart" element={<HistogramPage />} />
+              <Route
+                path="making-your-first-chart"
+                element={<WeatherDataProvider Consumer={LineChart} />}
```

Comment:
Looks good enough for now, but we might also wanna consider other ways to improve the WeatherDataProvider so we don't have to pass the charts like this every time.